File: frontend/cli/src/app/articulospedido/articulospedido.ts
import { Articulo } from "../articulos/articulo";
import { Factura } from "../facturas/factura";
import { Pedido } from "../pedidos/pedido";
import { Remito } from "../remitos/remito";

export interface ArticulosPedido {
  id: number;

  cantidad: number;

  precio: number;

  pedido: Pedido;

  articulo: Articulo;

  remito: Remito;

  factura: Factura;
}

File: backend/src/main/java/unpsjb/labprog/backend/model/ArticulosPedido.java
package unpsjb.labprog.backend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import jakarta.persistence.ManyToOne;
import jakarta.persistence.JoinColumn;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class ArticulosPedido {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    private int cantidad;
    private float precio;

    @ManyToOne
    @JoinColumn(name = "pedido_id")
    private Pedido pedido;

    @ManyToOne
    @JoinColumn(name = "articulo_id")
    private Articulo articulo;
    @ManyToOne
    @JoinColumn(name = "remito_id")
    private Remito remito;
    @ManyToOne
    @JoinColumn(name = "factura_id")
    private Factura factura;

}